lib/ngx/balancer.lua

```diff
@@ -15,22 +16,62 @@ local getfenv = getfenv
 local error = error
 local type = type
 local tonumber = tonumber
-
-
-ffi.cdef[[
-int ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
-    const unsigned char *addr, size_t addr_len, int port, char **err);
-
-int ngx_http_lua_ffi_balancer_set_more_tries(ngx_http_request_t *r,
-    int count, char **err);
-
-int ngx_http_lua_ffi_balancer_get_last_failure(ngx_http_request_t *r,
-    int *status, char **err);
-
-int ngx_http_lua_ffi_balancer_set_timeouts(ngx_http_request_t *r,
-    long connect_timeout, long send_timeout,
-    long read_timeout, char **err);
-]]
+local max = math.max
+local subsystem = ngx.config.subsystem
+local ngx_subsystem_lua_ffi_balancer_set_current_peer
+local ngx_subsystem_lua_ffi_balancer_set_more_tries
+local ngx_subsystem_lua_ffi_balancer_get_last_failure
+
+
+if subsystem == 'http' then
+    ffi.cdef[[
+    int ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
+        const unsigned char *addr, size_t addr_len, int port, char **err);
+
+    int ngx_http_lua_ffi_balancer_set_more_tries(ngx_http_request_t *r,
+        int count, char **err);
+
+    int ngx_http_lua_ffi_balancer_get_last_failure(ngx_http_request_t *r,
+        int *status, char **err);
+
+    int ngx_http_lua_ffi_balancer_set_timeouts(ngx_http_request_t *r,
+        long connect_timeout, long send_timeout,
+        long read_timeout, char **err);
+    ]]
+
+    ngx_subsystem_lua_ffi_balancer_set_current_peer =
+    C.ngx_http_lua_ffi_balancer_set_current_peer
+    ngx_subsystem_lua_ffi_balancer_set_more_tries =
+    C.ngx_http_lua_ffi_balancer_set_more_tries
+    ngx_subsystem_lua_ffi_balancer_get_last_failure =
+    C.ngx_http_lua_ffi_balancer_get_last_failure
+
+elseif subsystem == 'stream' then
+    ffi.cdef[[
+    int ngx_stream_lua_ffi_balancer_set_current_peer(
+        ngx_stream_lua_request_t *r,
+        const unsigned char *addr, size_t addr_len, int port, char **err);
+
+    int ngx_stream_lua_ffi_balancer_set_more_tries(ngx_stream_lua_request_t *r,
+        int count, char **err);
+
+    int ngx_stream_lua_ffi_balancer_get_last_failure(
+        ngx_stream_lua_request_t *r, int *status, char **err);
+
+    int ngx_stream_lua_ffi_balancer_set_timeouts(ngx_stream_lua_request_t *r,
+        long connect_timeout, long timeout, char **err);
+    ]]
+
+    ngx_subsystem_lua_ffi_balancer_set_current_peer =
+    C.ngx_stream_lua_ffi_balancer_set_current_peer
```

Comment:
Need more indentations on the continued line. Please fix other similar places as well.